schema.ts

```diff
@@ -0,0 +1,246 @@
+/**
+ * The top-level interface of a custom-elements.json file.
+ * 
+ * custom-elements.json documents all the elements in a single npm package,
+ * across all modules within the package. Elements may be exported from multiple
+ * modules with re-exports, but as a rule, elements in this file should be
+ * included once in the "canonical" module that they're exported from.
+ */
+export interface PackageDoc {
+  version: string;
+
+  /**
+   * An array of the modules this package contains.
+   */
+  modules: Array<ModuleDoc>;
+}
+
+export interface ModuleDoc {
+  path: string;
+
+  /**
+   * A markdown summary suitable for display in a listing.
+   */
+  summary?: string;
+
+  /**
+   * A markdown description of the module.
+   */
+  description?: string;
+
+  exports?: Array<ExportDoc>;
+}
+
+export type ExportDoc = ClassDoc|FunctionDoc|VariableDoc;
+
+/**
+ * A reference to an export of a module.
+ * 
+ * All references are required to be publically accessible, so the canonical
+ * representation of a refernce it the export it's available from.
+ */
+export interface Reference {
+  name: string;
+  package?: string;
+  module?: string;
+}
+
+export interface CustomElementDoc extends ClassDoc {
+  tagName: string;
+  /**
+   * The attributes that this element is known to understand.
+   */
+  attributes?: AttributeDoc[];
+
```

Comment:
having no "flattened" properties here makes is severely harder to list them for documentation?

e.g. I need to go to the class get all members and mixins and construct it myself... quite cumbersome :/

Reply:
I think I would also prefer a flat version of `custom-elements.json`. Moving the complexity to the "builder" of the file instead of the "consumer".

But there is a use case that would change my mind.

Reply:
We can definitely do flattened properties. In the Polymer Analyzer we flatten, but in each member indicate where they originated from: https://github.com/Polymer/tools/blob/master/packages/analyzer/src/analysis-format/analysis-format.ts#L327

We can do the same here, though I think that should be expanded from a string to a reference type.

This allows us to show where members are inherited from in the documentation viewer UI. Examples here: https://polymer-library.polymer-project.org/3.0/api/polymer-element

Reply:
Best of both worlds, I like that